File: test/fixtures/sample.go
package main

import "fmt"

func greet(name string) string {
	return "Hello, " + name
}

func validate(email string) bool {
	if len(email) > 0 {
		return true
	}
	return false
}

func processUser(name string, email string) {
	greeting := greet(name)
	fmt.Println(greeting)
	if validate(email) {
		saveUser(name, email)
	}
}

func saveUser(name string, email string) {
	fmt.Println("Saving user:", name, email)
}

func main() {
	processUser("Alice", "alice@test.com")
}

File: test/fixtures/sample.ts
function greet(name: string): string {
  return "Hello, " + name;
}

function validate(email: string): boolean {
  if (email.length > 0) {
    return true;
  }
  return false;
}

function processUser(name: string, email: string): void {
  const greeting = greet(name);
  console.log(greeting);
  if (validate(email)) {
    saveUser(name, email);
  }
}

function saveUser(name: string, email: string): void {
  console.log("Saving user:", name, email);
}

function main(): void {
  processUser("Alice", "alice@test.com");
}
